module/Decision/src/Decision/Service/Member.php

```diff
@@ -27,9 +28,47 @@ public function getMembershipInfo()
     }
 
     /**
+     *
+     */
+    public function findMemberByLidNr($lidnr)
+    {
+        return $this->getMemberMapper()->findByLidnr($lidnr);
+    }
+
+    /**
+     * Find a member by (part of) its name.
+     *
+     * @param string $name (part of) the name of a member
+     * @pre $name must be at least 3 characters
+     *
+     * @return array|null
+     */
+    public function findMemberByName($name)
+    {
+        if (strlen($name) < 3) {
+            throw new \Zend\Code\Exception\InvalidArgumentException(
+                $this->getTranslator()->translate('Name must be at least 3 characters')
+            );
+        }
+
+        if (!$this->isAllowed('search')) {
+            throw new \User\Permissions\NotAllowedException(
+                $this->getTranslator()->translate('Not allowed to search for members.')
+            );
+        }
+        $parts = explode(' ', $name);
+        if (count($parts) == 1) {
+            return $this->getMemberMapper()->findByName($name, $name);
+        } else {
+            return $this->getMemberMapper()->findByName($parts[0], last($parts));
+        }
```

Comment:
This wouldn't work well for people with multiple first names or a middle name (like 'van'). See my other comment for a DQL-based solution.